Utils/expect_common.dart

```diff
@@ -355,6 +355,44 @@ class Expect {
   static void runtimeIsNotType<T>(Object? o) {
     _checkType(_checkIs<T>, false, o);
   }
+
+  /// Checks that all elements of this `Iterable` have type `T`, not a subtype
+  /// of `T`. For example `<int>[1, 2, 3] is Iterable<num>` returns `true`, but
+  /// `iterableElementsRuntimeIs<num>(<int>[1, 2, 3])` will throw
+  /// `ExpectException`. Fails if `it` is not instance of `Iterable<T>`
+  static void iterableElementsRuntimeIs<T>(dynamic it) {
+    if (it is! Iterable<T>) {
+      throw ExpectException("Expected Iterable<$T> but got ${it.runtimeType}");
+    }
+    if (it.isNotEmpty) {
+      X combine<X>(X x, X _) => x;
+      try {
+        (it.reduce)(combine<T>);
+      } catch (e) {
+        throw ExpectException("Not $T in collection $it.");
+      }
+    }
+  }
+
+  /// Checks that all elements of this `Iterable` don't have type `T`, but may
+  /// be a subtype of `T`. For example
+  /// `iterableElementsRuntimeIsNot<num>([1, 2, 3])` is Ok, but
+  /// iterableElementsRuntimeIsNot<num>([1, 2, 3 as num]) throws an
+  /// `ExpectException`
```

Comment:
Same kind of issue here: We can't use the elements to test anything like "this iterable does not implement `Iterable<T>` for a given `T`".